lightning/src/chain/channelmonitor.rs

```diff
@@ -1888,7 +1895,10 @@ impl<Signer: WriteableEcdsaChannelSigner> ChannelMonitor<Signer> {
 							} else { false }
 						});
 						if !htlc_update_confd {
-							res.insert(source.clone(), htlc.clone());
```

Comment:
The only behavior change this achieves is *not including htlcs with non-null preimage*. I assume you do all this *add an extra return field which is not even used* and all to prepare for a future update?

If that's correct, a comment here would be useful. Commit message might work as well, but it's pretty heavy.

Reply:
So it still returns something which is never used in rust-lightning itself. The assumption is that this extra data may be useful for the external API?